server/src/fat/java/com/ibm/ws/lars/rest/ApiTest.java

```diff
@@ -133,6 +133,20 @@ public void testGetAllAssets() throws ClientProtocolException, IOException, Inva
     }
 
     @Test
+    public void testGetUnpublishedAssets() throws ClientProtocolException, IOException, InvalidJsonAssetException {
+        repository.addAssetNoAttachments(AssetUtils.getTestAsset());
+        Asset publishedAsset = repository.addAndPublishAssetNoAttachments(AssetUtils.getTestAsset());
+        AssetList assets = repository.doGetAllAssets();
+        assertEquals("Wrong number of assets", 2, assets.size());
+        RepositoryContext userRepository = RepositoryContext.toUserContext(repository);
+        AssetList userAssets = userRepository.doGetAllAssets();
+        userRepository.close();
+        assertEquals("Wrong number of assets", 1, userAssets.size());
+        assertEquals("The wrong asset was retrieved", publishedAsset.get_id(), userAssets.get(0).get_id());
+
+    }
+
+    @Test
```

Comment:
Probably need more tests here for checking that users can't access unpublished assets while filtering and searching